### PR TITLE
[CI:DOCS] Improve podman-save and podman-push documentation

### PR DESCRIPTION
**Pull request by amluto**
podman-save has awkward semantics with regard tagging the saved image in the output.  Improve the documentation, and suggest using podman push as an alternative.

Add a corresponding example to podman push's manpage.

See https://github.com/containers/podman/issues/20515

#### Does this PR introduce a user-facing change?

I think None.  It's user-facing in the sense that it updates documentation.  I don't think it needs a release note.  Certainly no action is required.

If a release note is required, I suggest:

```release-note
Updated podman-save documentation to suggest using podman push instead.
```

### CODE REVIEW

File: docs/source/markdown/podman-push.1.md.in [hunk 1]
```diff
@@ -40,6 +40,9 @@ $ sudo podman push myimage docker-daemon:docker.io/library/myimage:33
 
 # Push to a tarball in the OCI format
 $ podman push myimage oci-archive:/tmp/myimage
+
+# Push to a directory, with a relative path, in the OCI format with another tag
```
Comment (rhatdan) [inline docs/source/markdown/podman-push.1.md.in:44]: Add something about this being a good alternative to `podman save`

File: docs/source/markdown/podman-save.1.md.in [hunk 2]
```diff
@@ -39,6 +41,14 @@ An image format to produce, one of:
...
 
+If **oci-archive** or **oci-dir** saves an image specified by tag, then it will save it in the OCI Image with the same tag.  If **oci-archive** or **oci-dir** saves an image specified by hash, it will not tag it at all, which may make using the resulting image unnecessarily complicated.
+
+To explicity specify tag to be used when saving an image, use the **oci-archive:** or **oci::** schemes with **[podman-push(1)](podman-push.1.md)** instead.  For example:
+
+```
+$ podman push IMAGE oci:directoryname:tagname
+```
+
 #### **--help**, **-h**
```
Comment (rhatdan) [inline docs/source/markdown/podman-save.1.md.in:53]: Make sure you reference `podman push` in the See Also section below.

File: docs/source/markdown/podman-save.1.md.in [hunk 3]
```diff
@@ -22,6 +22,8 @@ Note: `:` is a restricted character and cannot be part of the file name.
 
 **podman save [OPTIONS] NAME[:TAG]**
 
+Most or all of **podman save**'s functionality is available, with more flexibility, in **[podman-push(1)](podman-push.1.md)**.
```
[the remark this answers is not in the document]
Comment (amluto) [reply docs/source/markdown/podman-save.1.md.in:25]: The comments in #20515 made it sounds like podman save is mostly for Docker compatibility.  I'll defer to people actually involved in podman.

File: docs/source/markdown/podman-push.1.md.in [hunk 4]
```diff
@@ -40,6 +40,9 @@ $ sudo podman push myimage docker-daemon:docker.io/library/myimage:33
...
+$ podman push myimage oci:directory:tag
```
[the remark this answers is not in the document]
Comment (amluto) [reply docs/source/markdown/podman-push.1.md.in:45]: I feel like adding a name like `quay.io/myrepo/myimage:mytag` makes the result more complicated.

**Comment by amluto** [comment]
Bah, I made a silly typo pushing an update and github interpreted that as closing the PR.

File: docs/source/markdown/podman-save.1.md.in [hunk 5]
```diff
@@ -39,6 +41,14 @@ An image format to produce, one of:
 | **oci-dir**        | A directory using the OCI Image Format                                       |
 | **docker-dir**     | **dir** transport (see **containers-transports(5)**) with v2s2 manifest type |
 
+If **oci-archive** or **oci-dir** saves an image specified by name, then it will save it in the OCI Image with the same name (i.e. an **org.opencontainers.image.ref.name** annotation will be written).  If **oci-archive** or **oci-dir** saves an image specified by hash, the image will not be given a name in the saved image, which may make using the resulting image with tools that expect named images difficult.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-save.1.md.in:44]:
```suggestion
If **oci-archive** or **oci-dir** saves an image specified by name, then it will keep it in the OCI Image with the same name (i.e., an **org.opencontainers.image.ref.name** annotation will be written).  If **oci-archive** or **oci-dir** saves an image specified by hash, the image will not be given a name in the saved image, which may make using the resulting image with tools that expect named images difficult.
```

Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-save.1.md.in:44]: I'm on the fence about keep/save.  The comma would be good though.

File: docs/source/markdown/podman-save.1.md.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-save.1.md.in:25]: I'd not include this

File: docs/source/markdown/podman-push.1.md.in [hunk 6]
```diff
@@ -28,7 +28,11 @@ $ podman push docker://quay.io/podman/stable
 
 # Push to a container registry with another tag
 $ podman push myimage quay.io/username/myimage
+```
+
+**podman push** can also be used as an alternative to **podman save**.  For example:
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-push.1.md.in:33]: I was looking for a wrap up note to the original "For example:".  I suggest this.

```suggestion
**podman push** can also be an alternative to **podman save**.  For instance, the following push commands save the image at the specified destination.
```

File: docs/source/markdown/podman-save.1.md.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (vrothberg) [reply docs/source/markdown/podman-save.1.md.in:25]: I think the statement is correct but also thin that the text above is sufficient to highlight that podman-push can be used to create archives as well.

**Review by vrothberg — COMMENTED** [review]
Great work, thanks!  We can merge once the comments are addressed.  I really like the description of the behavior when only a digest/hash is specified.

File: docs/source/markdown/podman-save.1.md.in (hunk 5 above)
Comment (amluto) [reply docs/source/markdown/podman-save.1.md.in:44]: I'll keep "save". Saying "keep" seems odd -- the image doesn't start out in the OCI Image at all, so the image isn't being kept.

**Comment by amluto** [comment]
I'm leaving the "Most or all of **podman save**'s functionality is available" part for now -- the other text that conveys that is farther down and maybe easy to miss.  But feel free to argue with me :)

**Comment by martinpitt** [comment]
Over night apparently quay.io was down, so all cockpit-revdeps tests failed with
```
Trying to pull quay.io/prometheus/busybox:latest...
Error: initializing source docker://quay.io/prometheus/busybox:latest: can't talk to a V1 container registry
```
It seems to be working again, so feel free to `/packit retry-failed` (or click "Re-run" [here](https://github.com/containers/podman/pull/20537/checks?check_run_id=18680659843))

File: docs/source/markdown/podman-push.1.md.in [hunk 7]
```diff
@@ -40,6 +44,9 @@ $ sudo podman push myimage docker-daemon:docker.io/library/myimage:33
 
 # Push to a tarball in the OCI format
 $ podman push myimage oci-archive:/tmp/myimage
+
+# Push to a directory in the OCI format with an explicitly specified tag
+$ podman push myimage oci:/tmp/myimage:tag
```
Comment (edsantiago) [inline docs/source/markdown/podman-push.1.md.in:49]: This is a counterintuitive situation for many users. Please add a note making it unambiguously clear that the destination directory is `/tmp/myimage` and that `tag` is an annotation:
```
# Push to a directory ...
$ podman push myimage oci:/tmp/myimage:myname/mytag
$ jq '.manifests[].annotations' /tmp/myimage/index.json
{
  "org.opencontainers.image.ref.name": "myname/mytag"
}
```

File: docs/source/markdown/podman-save.1.md.in [hunk 8]
```diff
@@ -39,6 +41,14 @@ An image format to produce, one of:
 | **oci-dir**        | A directory using the OCI Image Format                                       |
 | **docker-dir**     | **dir** transport (see **containers-transports(5)**) with v2s2 manifest type |
 
+If **oci-archive** or **oci-dir** saves an image specified by name, then it will save it in the OCI Image with the same name (i.e., an **org.opencontainers.image.ref.name** annotation will be written).  If **oci-archive** or **oci-dir** saves an image specified by hash, the image will not be given a name in the saved image, which may make using the resulting image with tools that expect named images difficult.
```
Comment (edsantiago) [inline docs/source/markdown/podman-save.1.md.in:44]: Awkward phrasing, I find it hard to parse. At the very least please differentiate **by name** vs **by hash**  to make it easier for a human reader.

File: docs/source/markdown/podman-save.1.md.in [hunk 9]
```diff
@@ -39,6 +41,14 @@ An image format to produce, one of:
...
+
+To explicity specify the name to be used in the saved image, use the **oci-archive:** or **oci::** schemes with **[podman-push(1)](podman-push.1.md)** instead.  For example:
```
Comment (edsantiago) [inline docs/source/markdown/podman-save.1.md.in:46]: Typo, "explicitly"

File: docs/source/markdown/podman-save.1.md.in [hunk 10]
```diff
@@ -39,6 +41,14 @@ An image format to produce, one of:
...
+
+```
+$ podman push IMAGE oci:directoryname:tagname
```
Comment (edsantiago) [inline docs/source/markdown/podman-save.1.md.in:49]: I have to insist that this is confusing: a nonexpert reader will have trouble figuring out what is an exact keyword and what is a mutable argument. See above for (IMO) clearer ways to say this.

**Review by edsantiago — CHANGES_REQUESTED** [review]
Thank you for your patience. Comments inline. Please also rebase to a current `main` and squash commits.
